helper_15.go

```diff
@@ -0,0 +1,45 @@
+// +build !go1.7
+
+/********************************
+*** Multiplexer for Go        ***
+*** Bone is under MIT license ***
+*** Code by CodingFerret      ***
+*** github.com/go-zoo         ***
+*********************************/
+
+package bone
+
+import (
+	"net/http"
+	"sync"
+)
+
+var globalVars = struct {
+	sync.RWMutex
+	v map[*http.Request]map[string]string
+}{v: make(map[*http.Request]map[string]string)}
+
+// GetAllValues return the req PARAMs
+func GetAllValues(req *http.Request) map[string]string {
+	globalVars.RLock()
+	values := globalVars.v[req]
+	globalVars.RUnlock()
+	return values
```

Comment:
Since maps are reference types, we risk concurrency issues if the user of this mutates the map, the most correct thing would be to copy everything when this is requested, but I actually think the perf tradeoff isn't worth it.